cf-oscore/src/main/java/org/eclipse/californium/oscore/OscoreOptionDecoder.java

```diff
@@ -179,7 +179,8 @@ public int getSequenceNumber() {
 			return 0;
 		}
 
-		byte[] partialIvInt = Arrays.copyOf(partialIV, Integer.SIZE / Byte.SIZE);
+		int padding = Integer.SIZE / Byte.SIZE - partialIV.length;
+		byte[] partialIvInt = OSSerializer.leftPaddingZeroes(partialIV, padding);
```

Comment:
Seems, this is somehow more complicated ... 
Have you considered the Californium utility class `DatagramReader` as well? With that it's:

```
return new DatagramReader(partialIV, false).read(partialIV.length * Byte.SIZE);
```

Reply:
Thanks for the tip! No I had not considered that one. Let me adjust the code to use that instead, test, and report back.

Reply:
Using the DatagramReader for this worked out nicely. I have now changed to using it instead in commit f488416cbb3b3933bde4d8d8453c0aedb86419a7